src/decompression.jl

```diff
@@ -286,3 +286,96 @@ function decompress_aux!(
     end
     return A
 end
+
+function decompress_aux!(
+    A::AbstractMatrix{R}, B::AbstractMatrix{R}, result::TreeSetColoringResult
+) where {R<:Real}
+    A .= zero(R)
+    S = get_matrix(result)
+    color = column_colors(result)
+    forest = result.tree_set.forest
+
+    ntrees = forest.internal.ngroups
+    for edge in forest.revmap
+        # ensure that all paths are compressed
+        find_root!(forest, edge)
+    end
```

Comment:
what is the complexity of this loop?

Reply:
The complexity of this loop is `nedges`.
The cost of `find_root!` is O(1).